galasa-managers-parent/galasa-managers-common-parent/dev.galasa.common/src/test/java/dev/galasa/SSLTLSContextNameTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.common;
+
+import org.junit.Test;
+
+import static org.junit.Assert.* ;
+import java.util.Properties;
+
+public class SSLTLSContextNameTest {
+    @Test
+    public void testNonIBMJVMGivesTLS12Context() throws Exception {
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"FakeJavaRuntimeVendorName" );
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "TLSv1.2";
+        assertEquals(expected,contextName);
+    }
+    @Test
+    public void testIBMJVM8GivesSSL_TLSv2Context() throws Exception {
+        // Given...
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"IBMJavaRuntimeVendorName" );
+        testProps.setProperty(SSLTLSContextName.JAVA_VERSION_PROPERTY, "8.0.16");
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "SSL_TLSv2";
+        assertEquals(expected,contextName);
+    }
+    
+    @Test
+    public void testIBMJVM11GivesTLSv12Context() throws Exception {
+        // Given...
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"IBMJavaRuntimeVendorName" );
+        testProps.setProperty(SSLTLSContextName.JAVA_VERSION_PROPERTY, "11.0.16.1+1");
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "TLSv1.2";
+        assertEquals(expected,contextName);
+    }
+
+    @Test
+    public void testSystemPropertiesContextIsSelected() throws Exception {
+        // Given...
+        System.out.println(System.getProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY));
+        System.out.println(System.getProperty(SSLTLSContextName.JAVA_VERSION_PROPERTY));
```

Comment:
This test will run through a different path depending on the caller's environment. Which may lead to tests working for you, but failing for others.